staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -1718,7 +1721,20 @@ func TestValidationExpressions(t *testing.T) {
 					if celValidator == nil {
 						t.Fatal("expected non nil validator")
 					}
-					errs, _ = celValidator.Validate(field.NewPath("root"), &s, tt.obj, tt.costBudget)
+					errs, _ = celValidator.Validate(field.NewPath("root"), &s, tt.obj, tt.costBudget, ctx)
+					for _, err := range errs {
+						if err.Type == field.ErrorTypeInternal && strings.Contains(err.Error(), "operation cancelled: actual cost limit exceeded") {
+							found = true
+						}
+					}
+					if !found {
+						t.Errorf("expect PerCostLimit exceed err but did not find")
+					}
+
+					// test context cancellation
+					evalCtx, cancel := context.WithTimeout(ctx, time.Microsecond)
+					defer cancel()
```

Comment:
maybe call cancel before even calling ContextEval to keep this test from flaking? Then so long as the input and frequency of the cancel check don't change, the test should be deterministic?

Reply:
cancel is called before calling ContextEval. The above line is to initialize evalCtx.